### DISCUSSION TITLE
Compilation error in parse.go

### DISCUSSION THREAD

**Discussion by gauravpahuja17** [moby/moby#48329] [General]
Facing compilation error when i upgrade package to v25.0.6
"
vendor/github.com/docker/cli/opts/parse.go:96:11: cannot use parts[0] (variable of type string) as container.RestartPolicyMode value in assignment
"

**Comment by laurazard** [comment]
There were likely breaking changes in the signature of the function you're calling. Instead of passing a `string` to represent the restart condition, you should pass a `container.RestartPolicyMode`. See: https://github.com/moby/moby/blob/49956b101c1abe7db6e9d7732a357c0b3352f58f/api/types/container/hostconfig.go#L283-L286
